### ISSUE TITLE
First run always lags behind

### ISSUE THREAD

**Issue by ghunti** [laurenz/pgreplay#26]
Hi.
I have a problem that I cannot explain and debug further, even though I've seen all closed issues here.

I have a CSV with 1 second of a log with 527 lines, connections, deallocates, prepares, and executes; all queries are SELECT.

When I start a new database and run the `pgreplay -r -d 1 -j -h <host> -W <pass>` against it, the process takes around 5 minutes to execute. If I run the same command a second time, it takes 1 second to complete.

At first, I was suspicious that this was due to some Postgresql cache and that the database needs to "warm" to start performing, so I've tried the same process with a log containing 1 hour's worth of queries. After 2 hours of running, only 12 minutes of the log were executed, so I stopped the process. I re-ran the same command, and the first 12 minutes of the log were completed within 12 minutes, but then it started to lag again.

I've counted the number of lines each second has because I've read on #18 that there seems to be a limit of 2000 queries per second, and while on some seconds I do have those numbers, of the vast majority, I'm far from it.

I've seen this problem with databases with the same or better ones than the original database.

Can you help me somehow understand this, or help me debug it?

I really appreciate any help you can provide.

**Comment by ghunti** [comment]
Hi @laurenz 
So, I've finally found the problem. I forgot to mention that I'm using an RDS restored from a snapshot.

Today I learned: "If you access data that hasn't been loaded yet, the DB instance immediately downloads the requested data from Amazon S3, and then continues loading the rest of the data in the background."

AWS calls this "lazy load", and it's documented here: https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/USER_RestoreFromSnapshot.html

So basically, I was using the RDS from the moment it was "available", but all my queries were waiting a long time for the data to be fetched from S3 the first time they executed.
I've solved this, by importing the SQL dump the "good old way" with psql
